### ISSUE TITLE
Unexpected Trilateration Results

### ISSUE THREAD

**Issue by fclinton** [ESPresense/ESPresense-companion#94]
Results shown for devices seems illogical just from looking at the distance circles

![ESPresense Companion Results (2)](https://user-images.githubusercontent.com/7285654/226089718-bf1527c2-f277-4c93-a913-7ce6287b8888.png)
![ESPresense Companion Results (3)](https://user-images.githubusercontent.com/7285654/226089719-e737e173-77ab-4a29-89e8-ceda62bb53ec.png)
![ESPresense Companion Results](https://user-images.githubusercontent.com/7285654/226089720-8231cbd2-6162-4241-b3ac-509a93395b4d.png)

**Comment by DTTerastar** [comment]
The algorithm uses the distances in order from closest to farthest.  The nearest distance is like 50% of the location.  Since there isn't anything close to that spot, and all of the other ones are very similar it got confused.  You need to have one node fairly close (1-3m) and some others on the corners and I think it'd get a better solution.

**Comment by DTTerastar** [comment]
(https://www.mdpi.com/1424-8220/20/24/7003) has an interesting post processing step when picking the nodes to use.  They pick 3 in order from closest to farthest but kick out nodes that aren't going to be useful (because they are too close or in the same line as the other nodes).  I'm going to try and work out how to do that.

**Comment by AlmostInteractive** [comment]
How did you show those distance circles?  I'd love to see that for my setup.  My triangulation is never closer than 3m from my actual position and that might help me debug why that is.

**Comment by fclinton** [comment]
> How did you show those distance circles? I'd love to see that for my setup. My triangulation is never closer than 3m from my actual position and that might help me debug why that is.

Hover your mouse on the beacon on the map

**Comment by AlmostInteractive** [comment]
My map only shows the nodes:
![image](https://user-images.githubusercontent.com/3893631/232349651-f5341375-36eb-4603-97e5-b8f9a67aac4b.png)

In the Devices I can see my phone, but it's not on the map:
![image](https://user-images.githubusercontent.com/3893631/232349742-c9a0f56f-e1c2-4ae4-a603-491f327f0d2c.png)

Did I setup the config wrong maybe?

**Comment by rhamblen** [comment]
![image](https://github.com/ESPresense/ESPresense-companion/assets/16073422/df047d3d-4531-4bad-9bd2-af2ce0099896)

So I finally got everything setup. And Im also getting unexpected results
I've got rooms and two floors. I have espresense nodes over two floor.
Now for testing, I just focused on a single floor.
Initially I saw the devices in the wrong places, thinking this was due to the number of sensors, I then change the second floor nodes so they are on first floor but higher. However this made no difference.

![image](https://github.com/ESPresense/ESPresense-companion/assets/16073422/f6036196-b3af-427d-8eaa-f3fe11246619)

Looking at the ipad, espresnese itself thinks it is connected to the frontroom sensor, but this isnt even showing as one of the three for the mapping sensors for companion.

What occurred to me was I have brick walls, and this will be impacting the distance. So maybe when we draw walls we need to provide a correction factor due to the wall, some manual value we can derive from testing.

And using the sensors over the many floors, will also be handy as people who dont have concrete floors will be able to take advantage of the 3 dimensions for mapping.

But myAnd using the sensors over the many floors, will also be handy as people who dont have concrete floors will be able to take advantage of the 3 dimensions for mapping. overwhelming view is that ESPresense companion is starting to look like a viable option to give a position in a room, but it clearly needs tweaking.
